packages/ses-ava/src/ses-ava-test.js

```diff
@@ -1,16 +1,59 @@
+/* global globalThis */
 import 'ses';
 
-const { stringify } = JSON;
-const { defineProperty, freeze } = Object;
-const { apply } = Reflect;
+/**
+ * Copied from the ses-shim's console-shim.js file, since the idea is that
+ * these communicate not by export import, but rather by convention plus
+ * feature testing. The following test is from there:
+ *
+ *`makeCausalConsoleFromLoggerForSesAva` is privileged because it exposes
+ * unredacted error info onto the `Logger` provided by the caller. It
+ * should not be made available to non-privileged code.
+ *
+ * Further, we consider this particular API choice to be experimental
+ * and may change in the future. It is currently only intended for use by
+ * `@endo/ses-ava`, with which it will be co-maintained.
+ *
+ * Thus, this `console-shim.js` makes `makeCausalConsoleFromLoggerForSesAva` for
+ * on `globalThis` which it *assumes* is the global of the start compartment,
+ * which is therefore allowed to hold powers that should not be available
+ * in constructed compartments. It makes it available as the value of a
+ * global property named by a registered symbol named
+ * `MAKE_CAUSAL_CONSOLE_FROM_LOGGER_KEY_FOR_SES_AVA`.
+ *
+ * Anyone accessing this, including `@endo/ses-ava`, should feature test for
+ * this and be tolerant of its absence. It may indeed disappear from later
+ * versions of the ses-shim.
+ */
+const MAKE_CAUSAL_CONSOLE_FROM_LOGGER_KEY_FOR_SES_AVA = Symbol.for(
+  'MAKE_CAUSAL_CONSOLE_FROM_LOGGER_KEY_FOR_SES_AVA',
+);
+
+const optMakeCausalConsoleFromLoggerForSesAva =
+  globalThis[MAKE_CAUSAL_CONSOLE_FROM_LOGGER_KEY_FOR_SES_AVA];
 
 /**
- * @typedef {(...args: unknown[]) => void} Logger
+ * TODO For some reason, the following declaration (with "at-" as "@")
+ * doesn't work well for either TS or typedoc. For TS it seems to type
+ * `VirtualConsole` as `any` in a vscode hover. For typedoc it results in
+ * errors.
+ *
+ * at-typedef {import('ses/console-tools.js').VirtualConsole} VirtualConsole
```

Comment:
This type reference error may be because of the rename to `ses/src/console-shim.js`. We may need to export `ses/console-shim.js` in `package.json` to make this internal type addressable. We should avoid reaching into `src` from other packages in general.

Reply:
Is there an explanation anywhere of these type scoping issues? I remain perpetually confused about these, and trying to infer them by example has only dug me into a deeper hold of confusion ;)

Reply:
Talked to @kriskowal . This is fine. Resolving.

Reply:
Given my continuing ignorance of type scoping rules + our joint aversion to unnecessary "exports":, I’m planning to proceed with https://github.com/endojs/endo/pull/2109#discussion_r1513654976 unfixed. I expect future waves of type improvement to catch such things, so I think the TODO comment is adequate for now. Resolving.

Reply:
Let’s merge without an entry in `"exports"`. It’s not needed for this feature and I like how you’ve structured the feature so that this would be easy to extend if we find a need.